### PR TITLE
added error msg to SaaS APIs

### PR DESCRIPTION
**Pull request by QianZhu**
1. improved error msg for SaaS create_table and create_index

### CODE REVIEW

File: python/lancedb/remote/table.py [hunk 1]
```diff
@@ -71,52 +71,86 @@ def create_index(
...
-            "index_type": index_type,
-            "metric_type": metric,
-            "index_cache_size": index_cache_size,
-        }
-        resp = self._conn._client.post(
-            f"/v1/table/{self._name}/create_index/", data=data
+        """create_index() is not supported on the LanceDB cloud"""
+        raise NotImplementedError(
+            "create_index() is not supported on the LanceDB cloud"
+            "We are working on adding this feature."
```
[the remark this answers is not in the document]
Comment (changhiskhan) [reply python/lancedb/remote/table.py:82]: right now is it hard coded to `vector` column?

File: python/lancedb/remote/db.py [hunk 2]
```diff
@@ -214,10 +215,12 @@ def create_table(
             raise ValueError("Either data or schema must be provided.")
         if embedding_functions is not None:
             raise NotImplementedError(
-                "embedding_functions is not supported for remote databases."
+                "embedding_functions is not supported on the LanceDB Cloud."
                 "Please vote https://github.com/lancedb/lancedb/issues/626 "
                 "for this feature."
             )
+        if mode is not None:
+            raise NotImplementedError("mode is not supported on the LanceDB Cloud.")
```
[the remark this answers is not in the document]
Comment (QianZhu) [reply python/lancedb/remote/db.py:223]: just a matter of getting to it

File: python/lancedb/remote/table.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (QianZhu) [reply python/lancedb/remote/table.py:82]: yes it is

**Review by changhiskhan — COMMENTED** [review]
ok i think the overall problem here is that LanceDB Cloud and LanceDB OSS are different in the API in meaningful ways. We can put a facade but now it means these warnings have to be everywhere. Or it means potentially the user migrating code from OSS will run into errors. Ultimately the question might be whether we optimize for each individual DX separately or optimize for the switchover.

**Comment by QianZhu** [comment]
> ok i think the overall problem here is that LanceDB Cloud and LanceDB OSS are different in the API in meaningful ways. We can put a facade but now it means these warnings have to be everywhere. Or it means potentially the user migrating code from OSS will run into errors. Ultimately the question might be whether we optimize for each individual DX separately or optimize for the switchover.

It is to optimize for the switchover - oss users might not pay attention to our cloud API interfaces so when they switch over, such messages will remind them of what's avail in cloud APIs. For cloud users, they prob won't even using those parameters to begin with.

File: python/lancedb/remote/table.py [hunk 3]
```diff
@@ -104,6 +108,28 @@ def create_index(
         ... )
         >>> table.create_index("L2", "vector") # doctest: +SKIP
         """
+
+        if num_partitions is not None:
+            raise NotImplementedError(
```
Comment (eddyxu) [inline python/lancedb/remote/table.py:113]: Can we just `log.warning` here?  This call can get through with these parameters, don't need to interrupt user's workflow.

**Review by eddyxu — APPROVED** [review]

**Pull request merged**